addon/-private/system/references/has-many.js

```diff
@@ -45,11 +48,11 @@ HasManyReference.prototype.ids = function() {
 };
 
 HasManyReference.prototype.meta = function() {
-  return this.hasManyRelationship.manyArray.meta;
+  return this.hasManyRelationship.meta;
 };
 
 HasManyReference.prototype.push = function(objectOrPromise) {
-  return Ember.RSVP.resolve(objectOrPromise).then((payload) => {
+  return resolve(objectOrPromise).then((payload) => {
     var array = payload;
```

Comment:
We have eventually the same problem with `belongsTo`. I think it's best to create an issue and make a separate PR to fix it (if it's really a problem). I have not changed anything significant in this PR regarding references.